### PR TITLE
add variables to SSgarbage allowing hard deletes to be disabled in vv for testing

### PR DESCRIPTION
**Pull request by MrsTonedOne**

### CODE REVIEW

File: code/controllers/subsystem/garbage.dm [hunk 1]
```diff
@@ -200,6 +203,10 @@ SUBSYSTEM_DEF(garbage)
 					to_chat(admin, "## TESTING: GC: -- [ADMIN_VV(D)] | [type] was unable to be GC'd --")
 				#endif
 				I.failures++
+				#ifndef REFERENCE_TRACKING
+				if (ignore_gc_failure)
+					continue
+				#endif
```
Comment (LemonInTheDark) [inline code/controllers/subsystem/garbage.dm:209]: why the ifndef?

Comment (MrStonedOne) [reply code/controllers/subsystem/garbage.dm:209]: the continue would keep code flow from hitting the if check below, and i didn't want to deal with these things conflicting, so i just made it so the compile options overrides it.

Comment (LemonInTheDark) [reply code/controllers/subsystem/garbage.dm:209]: aight

**Review by LemonInTheDark — APPROVED** [review]

**Comment by MrStonedOne** [comment]
I'll make this a config later.

**Pull request closed without merging**